CIPs/CIP-79/cip-79.md

```diff
@@ -0,0 +1,270 @@
+
+## DID Method Name
+
+The name string that shall identify this DID method is: `3`.
+
+A DID that uses this method MUST begin with the following prefix: `did:3`. Per the [DID specification](https://w3c.github.io/did-core/), this string MUST be in lowercase. The remainder of the DID, after the prefix, is specified below.
+
+## Method Specific Identifier
+
+There are two versions of 3IDs, both versions use a Ceramic document as a way to update the DID document. 3IDv1 is the most recent version, 3IDv0 is supported for legacy reasons. If you are creating new 3IDs you should always use 3IDv1.
+
+### 3IDv1
+
+The method specific identifier for 3IDv1 is simply a Ceramic DocID. The DocID used refers to the Ceramic document which contains the information needed to construct the DID Document upon resolution. DocIDs are encoded according to [CIP-59](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-59/CIP-59.md).
+
+```
+3idv1 = "did:3:<docId>"
+```
+
+#### Example
+
+```sh
+did:3:kjzl6cwe1jw149tlplc4bgnpn1v4uwk9rg9jkvijx0u0zmfa97t69dnqibqa2as
+```
+
+### 3IDv0
+
+The method specific identifier for 3IDv0 is a [CID](https://github.com/multiformats/cid) as produced by the [IPLD](https://github.com/ipld/specs) codec [dag-cbor](https://github.com/ipld/specs/blob/master/block-layer/codecs/dag-cbor.md).
+
+```
+3idv1 = "did:3:<cid>"
+```
+
+#### Example
+
+```sh
+did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi
+```
+
+## CRUD Operation Definitions
+
+In this section the CRUD operations for a 3ID DID are defined.
+
+### Create (Register)
+
+A `3` DID is created by simply creating a Ceramic [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document. The [`tile`](https://github.com/ceramicnetwork/CIP/blob/master/CIPs/CIP-8/CIP-8.md) document type takes a DID as the *controller*, and it's recommended that a `did:key` is used. The *controller* is the DID which is allowed to update the document. The *family* of the document is set to `3id`, and the *deterministic* flag is set to `true`.
+
+Now the content of the document should consist of a JSON object with one property *publicKeys*. The value of this property should be an object where the value is a [multicodec](https://github.com/multiformats/multicodec/) *base58btc* encoded public key, the key for any given key should be the last *15* characters of the encoded public key.
+
+The 3ID DID method supports any type of public key that can be encoded using multicodec which can easily be extended to support quantum resistant signature and encryption schemes in the future.
+
+#### Example
+
+Below you can see an example of how to create a 3ID using the Ceramic javascript api.
+
+```js
+const doc = await ceramic.createDocument({
+  metadata: {
+    controllers: ['did:key:zQ3shQNcackrTByiYaPGso1Nt7b6r1gSMg4XXBmavzvTMqX1h'], // secp256k1 did:key
+    family: '3id'
+  },
+  content: {
+    publicKeys: {
```

Comment:
Why are we duplicating part of the key between the key and the value?  Why not either just an array of the full public keys, or if we care about being able to quickly look up a specific key it could be a map with the full public key as the key, and something irrelevant like `true` as the value

Reply:
This basically allows someone to have a specific name for a key, even if that's not recommended.

Reply:
Also would be a pain to change it at this point.

Reply:
oh interesting.  I guess the spec does use the word "should" - not "must" - when describing the structure of this object.  And I suppose you're not mentioning that in the spec because you don't want to encourage it?  But if we don't want to encourage it, then why are we allowing it?  Shouldn't we pick a position and go either one way or the other?

Reply:
> Also would be a pain to change it at this point.

Yep, totally fair.  Just wanted to understand if this is here because we actually want it this way, or just because of historical decisions we no longer stand by.

Reply:
I don't feel strongly that we must pick a position.